api/peering_test.go

```diff
@@ -2,14 +2,155 @@ package api
 
 import (
 	"context"
+	"reflect"
 	"testing"
 	"time"
 
+	"github.com/hashicorp/serf/testutil/retry"
 	"github.com/stretchr/testify/require"
 
 	"github.com/hashicorp/consul/sdk/testutil"
 )
 
+func peerExistsInPeerListings(peer *Peering, peerings []*Peering) bool {
+	for _, aPeer := range peerings {
+		isEqual := (peer.PeerID == aPeer.PeerID) &&
+			(reflect.DeepEqual(peer.PeerCAPems, aPeer.PeerCAPems)) &&
+			(peer.PeerServerName == aPeer.PeerServerName) &&
+			(peer.Partition == aPeer.Partition) &&
+			(peer.Name == aPeer.Name) &&
+			(reflect.DeepEqual(peer.PeerServerAddresses, aPeer.PeerServerAddresses)) &&
+			(peer.State == aPeer.State) &&
+			(peer.CreateIndex == aPeer.CreateIndex) &&
+			(peer.ModifyIndex) == aPeer.ModifyIndex
+
+		if isEqual {
+			return true
+		}
+	}
+
+	return false
+}
+
+func TestAPI_Peering_Read(t *testing.T) {
+	t.Parallel()
+	c, s := makeClientWithCA(t)
+	defer s.Stop()
+	s.WaitForSerfCheck(t)
+	ctx := context.Background()
```

Comment:
Just noticed this (and in other tests) should have a timeout otherwise a bad network connection might cause this test to hang for a long time